pallets/gear/src/manager/task.rs

```diff
@@ -17,48 +17,31 @@
 // along with this program. If not, see <https://www.gnu.org/licenses/>.
 
 use crate::{
-    manager::ExtManager, weights::WeightInfo, Config, DbWeightOf, DispatchStashOf, Event, Pallet,
-    ProgramStorageOf, QueueOf, TaskPoolOf, WaitlistOf,
+    manager::ExtManager, weights::WeightInfo, Config, DispatchStashOf, Event, Pallet, QueueOf,
 };
 use alloc::string::ToString;
 use common::{
     event::{
-        MessageWokenRuntimeReason, MessageWokenSystemReason, ProgramChangeKind, RuntimeReason,
-        SystemReason, UserMessageReadSystemReason,
+        MessageWokenRuntimeReason, MessageWokenSystemReason, RuntimeReason, SystemReason,
+        UserMessageReadSystemReason,
     },
     paused_program_storage::SessionId,
     scheduler::*,
     storage::*,
-    ActiveProgram, Gas, Origin, PausedProgramStorage, Program, ProgramState, ProgramStorage,
+    Gas, Origin,
 };
 use core::cmp;
 use gear_core::{
-    code::MAX_WASM_PAGE_COUNT,
     ids::{CodeId, MessageId, ProgramId, ReservationId},
     message::{DispatchKind, ReplyMessage},
-    pages::{GEAR_PAGE_SIZE, WASM_PAGE_SIZE},
 };
 use gear_core_errors::{ErrorReplyReason, SignalCode};
-use sp_core::Get;
-use sp_runtime::Saturating;
 
 pub fn get_maximum_task_gas<T: Config>(task: &ScheduledTask<T::AccountId>) -> Gas {
     use ScheduledTask::*;
 
     match task {
-        PauseProgram(_) => {
-            // TODO: #3079
-            if <T as Config>::ProgramRentEnabled::get() {
-                let count =
-                    u32::from(MAX_WASM_PAGE_COUNT * (WASM_PAGE_SIZE / GEAR_PAGE_SIZE) as u16 / 2);
-                cmp::max(
-                    <T as Config>::WeightInfo::tasks_pause_program(count).ref_time(),
-                    <T as Config>::WeightInfo::tasks_pause_program_uninited(count).ref_time(),
-                )
-            } else {
-                DbWeightOf::<T>::get().writes(2).ref_time()
-            }
-        }
+        PauseProgram(_) => 0,
```

Comment:
these tasks may be removed atm

Reply:
for backward compatibility we shouldn't remove any variants of the enum. Nonetheless there are codec indexes attributes set I would leave the variants to avoid a mistakes in future.

Reply:
> for backward compatibility we shouldn't remove any variants of the enum. Nonetheless there are codec indexes attributes set I would leave the variants to avoid a mistakes in future.

Truth. Forgot about it